Popup component test

twang0323 opened the pull request.

twang0323 committed:
fixed format

shahdivyank commented:
Failing tests

twang0323 commented:
![image](https://github.com/acm-ucr/hackathon-website/assets/122486013/2adbbb16-8c19-42ad-afe0-e4b9fa9f4a80)

twang0323 committed:
fix tests

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,99 @@
+      visible: true,
+    };
+
+    const onClick = cy.stub();
```
Take a look at the tests that have a parent component and refer to the hooks being used

twang0323 committed:
deleted comments

twang0323 committed:
deleted comments

twang0323 commented:
![image](https://github.com/acm-ucr/hackathon-website/assets/122486013/1d0acebd-d0c9-4ce2-bc6d-76616b376235)

twang0323 committed:
have parent component

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,71 @@
+        color: "green",
+        visible: true,
+      });
+      const onClick = () => {};
```
use a cy.stub here

shahdivyank replied:
use stub

twang0323 replied:
const onClick = stub(); like this?

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,71 @@
+    // Click the cancel button
+    cy.get('[data-cy="cancel-button"]').click();
+
+    // Assert that the popup is no longer visible
```
remove comments

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,71 @@
+import { useState } from "react";
+
+describe("Popup Component", () => {
+  it("opens and displays the popup", () => {
```
```suggestion
  it("Open", () => {
```

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,71 @@
+    cy.contains("This is a test popup.").should("be.visible");
+  });
+
+  it("closes the popup when the close button is clicked", () => {
```
```suggestion
  it("Close", () => {
```

shahdivyank commented on src/components/admin/Popup.jsx:
```diff
@@ -4,38 +4,40 @@ import { LiaTimesSolid } from "react-icons/lia";
-              setPopup({ ...popup, visible: false });
-            }}
-          />
+    popup.visible && (
```
these changes should not be required

twang0323 replied:
before adding "popup.visible && ", if i call the component then it always exists, my not exist function is always failing

shahdivyank replied:
ahh this is because it was handled by the parent comp. u would have to do the same and make a parent comp which handles the existence of this child comp

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,71 @@
+import Popup from "@/components/admin/Popup";
```
Add a test for when the user clicks confirm

twang0323 committed:
fix onClick, removed comments, updated title, added confirm test

shahdivyank commented on hackathon-website:
remove this file

twang0323 replied:
what is this

shahdivyank commented on src/components/admin/Popup.jsx:
```diff
@@ -4,38 +4,40 @@ import { LiaTimesSolid } from "react-icons/lia";
 const Popup = ({ popup, onClick, setPopup, text }) => {
   console.log(popup);
```
remove console log

twang0323 commented:
![image](https://github.com/acm-ucr/hackathon-website/assets/122486013/8b7d9e9a-aad9-499b-92c7-5728b9ad2f6d)

twang0323 committed:
remove console log, add confirm test, add parent comp in test

twang0323 committed:
update

twang0323 committed:
fix onclick to stub

shahdivyank commented on cypress/component/admin/Popup.cy.jsx:
```diff
@@ -0,0 +1,97 @@
+      const onClick = cy.stub();
+
+      return (
+        <div>
```
make this a fragment

twang0323 commented:
![image](https://github.com/acm-ucr/hackathon-website/assets/122486013/d3eadd29-46a5-4d48-9e68-cccdece85372)

twang0323 committed:
change to fragment

shahdivyank commented:
make all of them fragments, only the first one is a fragment, also remove the extra file that is named hackathon-website

twang0323 commented:
mb i forgot the fragment about the other two tests, fixed now, and removed the extra hackathon-website
![image](https://github.com/acm-ucr/hackathon-website/assets/122486013/6021e4ea-4844-4c67-8aaa-7c3afc7ed4ba)

twang0323 committed:
changed all 3 to fragment and removed extra hackothon website file
